source/gui/settingsDialogs.py

```diff
@@ -2527,12 +2527,40 @@ def __init__(self, parent):
 		self.keyboardSupportInLegacyCheckBox.defaultValue = self._getDefaultValue(["terminals", "keyboardSupportInLegacy"])
 		self.keyboardSupportInLegacyCheckBox.Enable(winVersion.isWin10(1607))
 
-		# Translators: This is the label for a checkbox in the
-		#  Advanced settings panel.
-		label = _("Detect changes by c&haracter when supported")
-		self.useDMPWhenSupportedCheckBox = terminalsGroup.addItem(wx.CheckBox(self, label=label))
-		self.useDMPWhenSupportedCheckBox.SetValue(config.conf["terminals"]["useDMPWhenSupported"])
-		self.useDMPWhenSupportedCheckBox.defaultValue = self._getDefaultValue(["terminals", "useDMPWhenSupported"])
+		# Translators: This is the label for a combo box for selecting a
+		# method of detecting changed content in terminals in the advanced
+		# settings panel.
+		# Choices are automatic, character, and line.
+		diffAlgoComboText = _("C&hange detection preference:")
+		diffAlgoChoices = [
+			# Translators: A choice in a combo box in the advanced settings
+			# panel to have NVDA determine the method of detecting changed
+			# content in terminals automatically
+			_("automatic"),
+			# Translators: A choice in a combo box in the advanced settings
+			# panel to have NVDA detect changes in terminals
+			# by character when supported.
+			_("character"),
+			# Translators: A choice in a combo box in the advanced settings
+			# panel to have NVDA detect changes in terminals
+			# by line.
+			_("line")

```

Comment:
These options don't take into account the possibility of using an alternative "character based" approach. Eg an alternative dmp implementation. I think this only introduces a layer of obscurity. Given this is essentially a feature flag in the advanced settings, it can be more technical in nature. Once this proves itself it can either be turned into a user centric option or the configuration removed entirely.

Reply:
> These options don't take into account the possibility of using an alternative "character based" approach.

Why not? The alternative  approach could be called, say, "character (alternative)" in the UI and a new `diffAlgo` option could be added to the config spec (`dmpAlt`, say) with no need to upgrade the schema version.

Reply:
Except that is really confusing for developers and users alike. What I'm suggesting doesn't require a schema upgrade, the configSpec stays the same.

Reply:
OK, what option scheme are you proposing?

Reply:
@feerrenrut OK, how about showing the actual diff algorithms being used here? This should remove any ambiguity about what's going on, but would almost definitely require reading the docs to understand the setting. I suppose that's okay as it's an advanced option...